source/_posts/2015-10-09-the-slack-users-code.markdown

```diff
@@ -0,0 +1,258 @@
+channel names. #product-general isn't really any more descriptive than #product.
+
+#### Don't Create Private Channels
+
+You might think no one else cares. If they don't care, they don't have to join
+your channel. Private channels only block communication and discoverability.
+Your default assumption should be that all channels should be public. If you
+have something truly secret to discuss—something that even other employees of
+your company shouldn't see—consider that you're sharing anything you type into
+Slack with another company. If it's truly secret, consider sending PGP signed
+emails or only talking face-to-face. Much of what you do probably isn't actually
+that secret. Flowdock has [a great post about the dangers of private channels](http://blog.flowdock.com/2014/04/30/beware-of-private-conversations/).
+Also see [Zach Holman's fine writing about opt-in transparency](http://zachholman.com/posts/opt-in-transparency/).
+
+
+#### Don't Create Extraneous Channels
+
+You might think you're bothering people in another channel. For some kinds of
+channels (see below) it might be appropriate to create, but in general, you
+should wait until some sub-topic is routinely taking over a more general channel
+before you create a new one You don't want to get into a situation where there
+are more channels than people in your Slack team. Specifically, a team should
+not create a new channel for every individual problem they want to discuss. Keep
+things in as general a channel as folks can stand.
+
+
+#### Use Dashes in Channel Names
+
+Just for consistency. Do #austin-lunch-games not #austin_lunch_games or any
+other ridiculousness.
+
+
+#### Know What Kind of Channel You're Creating
+
+There are two broad categories of channel: Work-Related and Off-Topic.
+Work-Related channels are the ones that someone, somewhere in the organization
+is expected by their manager and team to be in and monitor actively. Off-Topic
+channels are everything else.
+
+
+#### Minimize Use of Bots & Other Toys
+
+Programmers, especially, can get enamoured of chat bots and chat integrations.
+As a programmer, I feel this, too. However, you have to carefully balance this
+with the cost it can have on real human communication. Slack is a tool primarily
+for humans to communicate with other humans. If computers are being too chatty
+in a room, the humans will have a hard time getting a word in edge-wise. And a
+chat history tends to be a pretty poor radiator of information. Rather than
+having engineers initiate and control a deploy from the chat room, consider
+having a different tool that does all that, and have only sparse or summary
+updates go to the team room.
+
+As for other toys, let us consider the cautionary tale of `/giphy`. The images
+that [Giphy](http://giphy.com/) serves up in response to any given query are
+actually relevant to the query so infrequently that it's a remarkable event when
```

Comment:
“…to any given query are so infrequently relevant that it’s remarkable when they communicate your intended meaning.”